### PR TITLE
add a test

### PR DESCRIPTION
**Pull request by thockin**
@stp-ip can you say what part of this test doesn't work for you?  It passes for me.

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/git-sync/18)

### CODE REVIEW

File: test.sh [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+git -C "$REPO" commit -qam "$TESTCASE 1"
+$GIT_SYNC \
+    --repo="$REPO" \
+    --branch=mybranch \
+    --root="$ROOT" \
+    --dest="link" > "$DIR"/log."$TESTCASE" 2>&1 &
+sleep 2
+assert_link_exists "$ROOT"/link
+assert_file_exists "$ROOT"/link/file
+assert_file_eq "$ROOT"/link/file "$TESTCASE 1"
```
Comment (stp-ip) [inline test.sh]: I think the issue doesn't come up when the latest revision is on the pulled branch, but when a new rev is pushed to another branch. Therefore adding the commit to the master branch should update the repo, even so it shouldn't be updated as the commit is from another branch. The test fails with the change. Please correct me, if I misinterpreted the test.

**Comment by thockin** [comment]
Interesting.  Ok.  I am still thinking about it. I can repro now.

On Oct 27, 2016 11:08 PM, "Michael Grosser" <notifications@github.com>
wrote:

*@stp-ip* commented on this pull request.
------------------------------

In test.sh <https://github.com/kubernetes/git-sync/pull/18>:

> +pass
+
+# Test branch syncing
+testcase "branch-sync"
+git -C "$REPO" checkout -q -b mybranch
+echo "$TESTCASE 1" > "$REPO"/file
+git -C "$REPO" commit -qam "$TESTCASE 1"
+$GIT_SYNC \
+    --repo="$REPO" \
+    --branch=mybranch \
+    --root="$ROOT" \
+    --dest="link" > "$DIR"/log."$TESTCASE" 2>&1 &
+sleep 2
+assert_link_exists "$ROOT"/link
+assert_file_exists "$ROOT"/link/file
+assert_file_eq "$ROOT"/link/file "$TESTCASE 1"

I think the issue doesn't come up when the latest revision is on the pulled
branch, but when a new rev is pushed to another branch. Therefore adding
the commit to the master branch should update the repo, even so it
shouldn't be updated as the commit is from another branch. The test fails
with the change. Please correct me, if I misinterpreted the test.

—
You are receiving this because you authored the thread.

Reply to this email directly, view it on GitHub
<https://github.com/kubernetes/git-sync/pull/18>, or mute the thread

**Comment by thockin** [comment]
pushed a more comprehensive test now

**Comment by thockin** [comment]
Assigning to Mike since @stp-ip doesn't have access, but maybe he should.  If he wanted to help maintain this...?

**Comment by thockin** [comment]
self-merging to make progress.

**Comment by stp-ip** [comment]
I can definitely help with reviews and issues as a first step so we have another set of eyes available. cc @thockin

**Comment by stp-ip** [comment]
Not seeing the more comprehensive test. Seems like the push wasn't available, when the merge occurred or did I misinterpret your comment @thockin?

**Comment by thockin** [comment]
The more comprehensive test checked across branches with tags and branches.

**Comment by thockin** [comment]
invite sent!

**Pull request merged**